### PR TITLE
[math-input] 🧹 Misc cleanups

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:

A few small cleanups: 
  * Move TabbarItemType to types.js to remove cyclic import
  * Add missing storybook/addon-knobs dependency
  * ESLint --fix and fix a few small flow issues
  * Fix dislay typo in tabbar item styles
  * We don't support @flow in multi-line comments /* */
  * math-input is no longer 'new'

Issue: LP-11432

## Test plan:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "@storybook/addon-a11y": "^6.4.19",
     "@storybook/addon-actions": "^6.4.19",
     "@storybook/addon-essentials": "^6.4.19",
+    "@storybook/addon-knobs": "^6.4.0",
```
Comment (jeremywiebe) [inline package.json:36]: We should migrate the `math-input` stories to `@storybook/addon-controls`, but I'll leave that for when we get Storybook going in this repo.

File: packages/math-input/src/components/keypad/index.js [hunk 2]
```diff
@@ -41,7 +37,7 @@ export default class PreAlgebraKeypad extends React.Component<Props, State> {
     };
     render(): React.Node {
         const {selectedPage} = this.state;
-        const {onClickKey, preAlgebra} = this.props;
```
Comment (jeremywiebe) [inline packages/math-input/src/components/keypad/index.js:44]: `preAlgebra` isn't used in this function.

File: packages/math-input/src/components/tabbar/item.js [hunk 3]
```diff
@@ -34,7 +36,7 @@ const styles = StyleSheet.create({
         boxSizing: "border-box",
         border: `1px solid transparent`,
         borderRadius: 2,
-        dislay: "flex",
+        display: "flex",
```
Comment (jeremywiebe) [inline packages/math-input/src/components/tabbar/item.js:39]: I wonder if I should leave this typo in (or delete the line entirely). Obviously this has been not actually applying `display: "flex"` for a while... so I'm not sure this'll cause a regression by fixing it.

File: packages/math-input/src/components/keypad/index.js [hunk 4]
```diff
@@ -1,8 +1,5 @@
 //@flow
-import Clickable from "@khanacademy/wonder-blocks-clickable";
-import Color from "@khanacademy/wonder-blocks-color";
 import {View} from "@khanacademy/wonder-blocks-core";
-import {StyleSheet} from "aphrodite";
```
Comment (kevinbarabash) [inline packages/math-input/src/components/keypad/index.js:5]: All of this stuff just wasn't being used?  🙃

File: packages/math-input/src/components/tabbar/item.js (hunk 3 above)
Comment (kevinbarabash) [reply packages/math-input/src/components/tabbar/item.js:39]: I guess we'll find out when QE regression tests things.   Was this caught by Flow?

File: yarn.lock [hunk 5]
```diff
@@ -1100,7 +1100,7 @@
     core-js-pure "^3.20.2"
     regenerator-runtime "^0.13.4"
 
-"@babel/runtime@^7.0.0", "@babel/runtime@^7.1.2", "@babel/runtime@^7.10.2", "@babel/runtime@^7.10.4", "@babel/runtime@^7.12.1", "@babel/runtime@^7.12.13", "@babel/runtime@^7.12.5", "@babel/runtime@^7.13.10", "@babel/runtime@^7.13.8", "@babel/runtime@^7.14.8", "@babel/runtime@^7.15.4", "@babel/runtime@^7.16.3", "@babel/runtime@^7.17.8", "@babel/runtime@^7.3.1", "@babel/runtime@^7.5.0", "@babel/runtime@^7.5.5", "@babel/runtime@^7.7.2", "@babel/runtime@^7.7.6", "@babel/runtime@^7.8.4", "@babel/runtime@^7.8.7", "@babel/runtime@^7.9.2":
+"@babel/runtime@^7.0.0", "@babel/runtime@^7.1.2", "@babel/runtime@^7.10.2", "@babel/runtime@^7.10.4", "@babel/runtime@^7.12.1", "@babel/runtime@^7.12.13", "@babel/runtime@^7.12.5", "@babel/runtime@^7.13.10", "@babel/runtime@^7.13.8", "@babel/runtime@^7.14.8", "@babel/runtime@^7.15.4", "@babel/runtime@^7.16.3", "@babel/runtime@^7.17.8", "@babel/runtime@^7.3.1", "@babel/runtime@^7.4.4", "@babel/runtime@^7.5.0", "@babel/runtime@^7.5.5", "@babel/runtime@^7.7.2", "@babel/runtime@^7.7.6", "@babel/runtime@^7.8.4", "@babel/runtime@^7.8.7", "@babel/runtime@^7.9.2":
```
Comment (kevinbarabash) [inline yarn.lock:1103]: 😂

**Review by kevinbarabash — APPROVED** [review]
LGTM 🚀

File: packages/math-input/src/components/keypad/index.js (hunk 4 above)
Comment (jeremywiebe) [reply packages/math-input/src/components/keypad/index.js:5]: Correct. It was all greyed out and was triggering our ESLint `no-unused` rule.

**Pull request merged**